### PR TITLE
Fix #3133 - Implement GPC (DOM only)

### LINKED ISSUE #3133: Implement GPC (dom only) - Part 1
From: https://github.com/brave/brave-browser/issues/11917

Add a navigator.globalPrivacyControl end point in the DOM, that always returns true (boolean, not string). No UX or shields integration is needed. No state is set or maintained.

## Test Case

Navigate to https://webtest-pj.herokuapp.com/gpc.html and verify that navigator.globalPrivacyControl is true for both the mainframe and the iframe.

### PR DESCRIPTION
**Pull request by jumde**
## Summary of Changes

This pull request fixes #3133 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
1. Navigate to https://webtest-pj.herokuapp.com/gpc.html and verify that `navigator.globalPrivacyControl` is true for both the mainframe and the iframe.

## Screenshots:
Screen Shot 2021-01-15 at 12 41 55 PM

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by kylehickinson — APPROVED** [review]

**Review by pes10k — APPROVED** [review]
Looks great w/ configurable change!

**Comment by pes10k** [comment]
Added tests for QA here https://dev-pages.brave.software/dom-properties.html

File: Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/gpc.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* vim: set ts=2 sts=2 sw=2 et tw=80: */
```
Comment (iccub) [inline Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/gpc.js:1]: not a big deal but i am not sure whether we should version vim settings

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Just piggy backing on @iccub's comment, can we please remove the changes to the files: `AllFramesAtDocumentEnd.js`, `MainFrameAtDocumentEnd.js` and `MainFrameAtDocumentStart.js`. Not sure why these files are changed when you only added a script to all frames start

**Review by kylehickinson — APPROVED** [review]
Discussed on Slack, will deal with webpack'd files in a separate commit as it seems to be generating different files regardless of folder the new JS is added to.

**Pull request merged**